src/helix/search/search_categories.rs

```diff
@@ -74,7 +76,46 @@ impl Request for SearchCategoriesRequest {
     const SCOPE: &'static [twitch_oauth2::Scope] = &[];
 }
 
-impl RequestGet for SearchCategoriesRequest {}
+impl RequestGet for SearchCategoriesRequest {
+    fn parse_response(
+        request: Option<Self>,
+        uri: &http::Uri,
+        response: http::Response<Vec<u8>>,
+    ) -> Result<
+        helix::Response<Self, <Self as helix::Request>::Response>,
+        helix::HelixRequestGetError,
+    >
+    where
+        Self: Sized,
+    {
+        let text = std::str::from_utf8(&response.body()).map_err(|e| {
+            helix::HelixRequestGetError::Utf8Error(response.body().clone(), e, uri.clone())
+        })?;
+        //eprintln!("\n\nmessage is ------------ {} ------------", text);
+        if let Ok(helix::HelixRequestError {
+            error,
+            status,
+            message,
+        }) = serde_json::from_str::<helix::HelixRequestError>(&text)
+        {
+            return Err(helix::HelixRequestGetError::Error {
+                error,
+                status: status.try_into().unwrap_or(http::StatusCode::BAD_REQUEST),
+                message,
+                uri: uri.clone(),
+            });
+        }
+        let response: helix::InnerResponse<Option<_>> =
```

Comment:
Think I want to look over the possibility of replacing `HelixRequestError` and `InnerResponse` as deser targets for less code-duplication.